packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualCreator.sol

```diff
@@ -65,7 +69,21 @@ contract PerpetualCreator is ContractCreator, Testable, Lockable {
      * @return address of the deployed contract.
      */
     function createPerpetual(Params memory params) public nonReentrant() returns (address) {
-        address derivative = PerpetualLib.deploy(_convertParams(params));
+        // Create a new synthetic token using the params.
+        require(bytes(params.syntheticName).length != 0, "Missing synthetic name");
+        require(bytes(params.syntheticSymbol).length != 0, "Missing synthetic symbol");
+        TokenFactory tf = TokenFactory(tokenFactoryAddress);
```

Comment:
Is there a reason we initiate the `TokenFactory` within the function, rather than storing it in the constructor? For example we could have a public instance of `tokenFactory` that is init within the constructor and re-used here. Not really a huge issue to have it like this, just seems to be unnecessary variable initalization.

Reply:
No reason, we can make this slightly less useless and just store `tokenFactory` in the state.

Reply:
Actually, did you see that `tokenFactoryAddress` is stored in the constructor? So you'd also suggest we store a `private TokenFactory tf` in the state?

Reply:
ok cool. I'm happy for it to stay where it is now & don't really think it makes too much difference. Was more curious from a bytecode perspective but we are not pushing any limits in the `PerpetualCreator` so probs fine.